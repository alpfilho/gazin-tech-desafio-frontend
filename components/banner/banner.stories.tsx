import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Banner } from './banner';

export default {
	title: 'Home/Banner',
	component: Banner,
	parameters: {
		layout: 'fullscreen'
	}
} as ComponentMeta<typeof Banner>;

export const Default: ComponentStory<typeof Banner> = (args) => <Banner {...args} />;

Default.args = {
	movies: [
		{
			id: 414906,
			title: 'Batman',
			overview:
				'Bruce Wayne é um jovem bilionário da cidade de Gotham City, Nova Jersey, que também age secretamente como o vigilante noturno Batman após o assassinato dos seus pais.',
			backdrop: 'http://image.tmdb.org/t/p/original/5P8SmMzSNYikXpxil6BYzJ16611.jpg',
			rating: 8.1
		},
		{
			id: 634649,
			title: 'Homem-Aranha: Sem Volta Para Casa',
			overview:
				'Peter Parker é desmascarado e não consegue mais separar sua vida normal dos grandes riscos de ser um super-herói. Quando ele pede ajuda ao Doutor Estranho, os riscos se tornam ainda mais perigosos, e o forçam a descobrir o que realmente significa ser o Homem-Aranha.',
			backdrop: 'http://image.tmdb.org/t/p/original/iQFcwSGbZXMkeyKrxbPnwnRo5fl.jpg',
			rating: 8.3
		},
		{
			id: 476669,
			title: "King's Man: A Origem",
			overview:
				'Quando uma série dos piores tiranos e gênios do crime da história se juntam para criar uma guerra que aniquilará milhões, um homem e seu pupilo precisam correr contra o tempo para pará-los.',
			backdrop: 'http://image.tmdb.org/t/p/original/oQPbZ5e6J9fuAyv4Gl0mMZMIyPI.jpg',
			rating: 7
		},
		{
			id: 823625,
			title: 'Agente das Sombras',
			overview:
				'Confiança, identidade e o perigo do poder sem controle levam um agente secreto ao limite. Travis Block (Liam Neeson) vive e luta nas sombras. Um “consertador” autônomo do governo, Block é um homem perigoso cujas atribuições incluíam extrair agentes de situações ocultas. Quando Block descobre que um programa sombrio chamado Operação Unidade está abatendo cidadãos comuns por razões conhecidas apenas pelo seu supervisor, o chefe do FBI, Robinson (Aidan Quinn), ele pede a ajuda de uma jornalista (Raver-Lampman), mas seu passado e presente colidem quando seu filha e neta são ameaçadas. Agora Block precisa resgatar as pessoas que ama e expor a verdade para uma chance de redenção. Nada nem ninguém está seguro quando os segredos estão escondidos nas sombras.',
			backdrop: 'http://image.tmdb.org/t/p/original/2hGjmgZrS1nlsEl5PZorn7EsmzH.jpg',
			rating: 5.1
		},
		{
			id: 511809,
			title: 'Amor, Sublime Amor',
			overview:
				'Dois jovens de gangues rivais de Nova York se apaixonam, mas as tensões entre seus respectivos amigos se transformam em tragédia. Remake do filme de 1961.',
			backdrop: 'http://image.tmdb.org/t/p/original/opM1iONReaoQSn0soOgCtDOGdnA.jpg',
			rating: 7.4
		},
		{
			id: 646385,
			title: 'Pânico',
			overview:
				'Vinte e cinco anos após uma série de crimes brutais chocar a tranquila Woodsboro, um novo assassino se apropria da máscara de Ghostface e começa a perseguir um grupo de adolescentes para trazer à tona segredos do passado mortal da cidade.',
			backdrop: 'http://image.tmdb.org/t/p/original/usaZV7KB6Man9Rm9TyDAeQf7uVD.jpg',
			rating: 6.8
		},
		{
			id: 836009,
			title: 'Contra o Gelo',
			overview:
				'Dois homens lutam pela sobrevivência durante uma expedição à Groenlândia em busca de um mapa perdido. Baseado em uma história real ocorrida na Dinamarca em 1909.',
			backdrop: 'http://image.tmdb.org/t/p/original/bnMKMhGJuvELSjRp8KpkuDsBDw2.jpg',
			rating: 7
		},
		{
			id: 833425,
			title: 'Sem Saída',
			overview:
				'Darby Thorne é uma mulher que viaja devido a uma emergência familiar, fica presa em nevasca e é forçada a se abrigar numa área de descanso na estrada com um grupo de desconhecidos. Ao topar com uma garota sequestrada numa van, ela encara uma terrível luta de vida ou morte para descobrir quem é o sequestrador.',
			backdrop: 'http://image.tmdb.org/t/p/original/33wnBK5NxvuKQv0Cxo3wMv0eR7F.jpg',
			rating: 6.3
		},
		{
			id: 591120,
			title: 'Os Piratas: Em Busca do Tesouro Perdido',
			overview:
				'Durante a Dinastia Joseon, uma corajosa tripulação de piratas e bandidos luta contra adversidades, pistas intrigantes e rivais em busca do ouro real perdido no mar.',
			backdrop: 'http://image.tmdb.org/t/p/original/pwpw1veVNW2Sls5sGbA9mqMYN49.jpg',
			rating: 6.6
		},
		{
			id: 632727,
			title: 'O Massacre da Serra Elétrica: O Retorno de Leatherface',
			overview:
				'Nesta sequência, um grupo de jovens decide dar vida nova a uma cidade fantasma no Texas, mas acaba se deparando com Leatherface, o famoso maníaco da serra elétrica.',
			backdrop: 'http://image.tmdb.org/t/p/original/aTSA5zMWlVFTYBIZxTCMbLkfOtb.jpg',
			rating: 5.1
		}
	]
};
