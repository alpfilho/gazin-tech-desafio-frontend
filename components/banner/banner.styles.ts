import styled from 'styled-components';
import { motion } from 'framer-motion';

import { Container, Content } from 'components/contentContainer';

import { Rating } from 'components/tmdbRating/tmdbRating.styles';
import { headerHeight } from 'components/header/header.styles';

export const BannerContainer = styled.section`
	width: 100%;
	height: calc(100vh - ${headerHeight});

	position: relative;
	isolation: isolate;

	background-image: url('/images/no_backdrop_image.png');
	background-size: cover;
	background-position-y: center;
	background-position-x: center;

	overflow: hidden;
`;

export const BannerContent = styled.div`
	position: relative;
	z-index: 1;
	width: 100%;
	height: 100%;

	${Container} {
		width: 100%;
		height: 100%;
		padding-left: 14em;
		padding-right: 14em;

		@media screen and (max-width: 1100px) {
			padding-left: 4em;
			padding-right: 4em;
		}

		@media screen and (max-width: 768px) {
			padding-left: 2em;
			padding-right: 2em;
		}

		${Content} {
			width: 100%;
			height: 100%;

			display: flex;
			flex-direction: column;
			justify-content: flex-start;
		}
	}
`;

export const RatingContainer = styled.div`
	padding-top: 4em;
	display: flex;
	justify-content: flex-end;
	align-items: flex-start;
	height: 70%;

	@media screen and (max-width: 1100px) {
		height: 60%;
		justify-content: center;
		padding-top: 2em;
	}

	@media screen and (max-width: 768px) {
		padding-top: 1em;
	}

	@media screen and (max-height: 600px) {
		height: 50%;
	}

	${Rating} {
		color: ${({ theme }) => theme.colors.white};
	}
`;

export const TextContainer = styled.div`
	position: relative;
	isolation: isolate;
	height: 40%;
	padding-top: 2em;
	padding-bottom: 2em;
	width: 100%;
	filter: drop-shadow(0px 2px 4px rgba(0, 0, 0, 0.33));
	color: ${({ theme }) => theme.colors.white};

	@media screen and (max-width: 1100px) {
		height: 40%;
	}

	@media screen and (max-width: 768px) {
		height: 20%;
	}
`;

export const BackgroundContainer = styled.div`
	position: absolute;
	z-index: 0;

	left: 0;
	right: 0;
	bottom: 0;
	top: 0;
`;

export const BackgroundContent = styled.div`
	isolation: isolate;
	position: relative;
	width: 100%;
	height: 100%;
	overflow: hidden;
`;

export const BackgroundImage = styled(motion.img)<{ backgroundimage: string }>`
	position: absolute;
	// por algum motivo as imagens sempre são retornadas com uma borda branca
	// fiz esta "gambiarra" para não aparecer
	left: -2px;
	right: -2px;
	bottom: -2px;
	top: -2px;
	width: calc(100% + 4px);
	height: calc(100% + 4px);

	background-image: url(${(props) => props.backgroundimage});
	background-size: cover;
	background-position-y: center;
	background-position-x: center;

	will-change: opacity;
`;

export const BackgroundOverlay = styled.div`
	position: absolute;
	z-index: 2;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	width: 100%;
	height: 100%;
	background-color: rgba(0, 0, 0, 0.33);
`;

export const TextBackground = styled.div`
	position: absolute;
	z-index: 3;
	bottom: 0;
	top: auto;
	left: 0;
	right: 0;
	width: 100%;
	height: 66%;
	@media screen and (max-width: 1100px) {
		height: 80%;
	}
	background: linear-gradient(0deg, rgba(0, 0, 0, 1) 0%, rgba(0, 0, 0, 0) 100%);
`;
