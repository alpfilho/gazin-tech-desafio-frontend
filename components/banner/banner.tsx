import React, { useCallback, useState, useRef, useEffect } from 'react';
import Link from 'next/link';

import type { BannerMovieData } from 'lib/types';

import { SliderControl } from 'components/sliderControl';
import { ContentContainer } from 'components/contentContainer';
import { TmdbRating } from 'components/tmdbRating';

import { TextContent } from './textContent';

import {
	BackgroundContainer,
	BackgroundContent,
	BackgroundImage,
	BackgroundOverlay,
	BannerContainer,
	BannerContent,
	RatingContainer,
	TextBackground,
	TextContainer
} from './banner.styles';

const loopInterval = 4000;

export const Banner: React.FC<{
	movies: BannerMovieData[];
}> = ({ movies }) => {
	const prevIndexRef = useRef(0);
	const [activeSlide, setSlideIndex] = useState(0);

	const goForwards = useCallback(() => {
		setSlideIndex((prevSlide) => {
			prevIndexRef.current = prevSlide;
			const nextSlide = prevSlide + 1;

			if (nextSlide > movies.length - 1) {
				return 0;
			}

			return nextSlide;
		});
	}, [movies.length]);

	const goBackwards = useCallback(() => {
		setSlideIndex((prevSlide) => {
			prevIndexRef.current = prevSlide;
			const nextSlide = prevSlide - 1;

			if (nextSlide < 0) {
				return movies.length - 1;
			}

			return nextSlide;
		});
	}, [movies.length]);

	const loopControlRef = useRef<{ timeoutID: NodeJS.Timeout | undefined; waitTime: number }>({
		timeoutID: undefined,
		waitTime: loopInterval
	});

	/**
	 * Loop
	 */
	const startTimer = useCallback(() => {
		const loopControl = loopControlRef.current;

		if (loopControl.timeoutID) {
			clearTimeout(loopControl.timeoutID);
		}

		loopControl.timeoutID = setTimeout(() => {
			goForwards();
		}, loopInterval);
	}, [goForwards]);

	useEffect(() => {
		startTimer();

		const loopControl = loopControlRef.current;

		return () => {
			if (loopControl.timeoutID) {
				clearTimeout(loopControl.timeoutID);
			}
		};
	}, [activeSlide, startTimer]);

	return (
		<BannerContainer>
			<SliderControl type="left" controlFn={goBackwards} color="white" show={true} />

			<BannerContent>
				<ContentContainer>
					<RatingContainer>
						{movies[activeSlide] && <TmdbRating rating={movies[activeSlide].rating} />}
					</RatingContainer>
					<Link href={`/movie-detail/${movies[activeSlide].id}`} passHref>
						<TextContainer>
							{movies.map(({ id, title, overview }, index) => {
								return (
									<TextContent
										key={id}
										itemIndex={index}
										title={title}
										overview={overview}
										activeSlide={activeSlide}
										slideLength={movies.length}
										prevIndexRef={prevIndexRef}
									/>
								);
							})}
						</TextContainer>
					</Link>
				</ContentContainer>
			</BannerContent>

			<BackgroundContainer>
				<BackgroundContent>
					{movies.map(({ id, backdrop }) => {
						const match = movies[activeSlide].id === id;
						return (
							<BackgroundImage
								key={id}
								backgroundimage={backdrop}
								initial={false}
								animate={{ opacity: match ? 1 : 0, zIndex: match ? 1 : 0 }}
								transition={{
									opacity: {
										type: 'tween',
										duration: 0.33,
										ease: 'linear',
										delay: match ? 0 : 0.33
									}
								}}
							/>
						);
					})}
					<BackgroundOverlay />
					<TextBackground />
				</BackgroundContent>
			</BackgroundContainer>

			<SliderControl type="right" controlFn={goForwards} color="white" show={true} />
		</BannerContainer>
	);
};
