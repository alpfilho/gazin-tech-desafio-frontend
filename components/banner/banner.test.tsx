import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './banner.stories';

const { Default } = composeStories(stories);

describe('Banner', () => {
	it('render a banner', () => {
		render(<Default />);

		const movie1Title = screen.getByText('Batman');
		const movie2Title = screen.getByText('Homem-Aranha: Sem Volta Para Casa');
		const movie3Title = screen.getByText("King's Man: A Origem");

		expect(movie1Title).toBeInTheDocument();
		expect(movie1Title).toBeVisible();
		expect(movie2Title).toBeInTheDocument();
		expect(movie2Title).toBeInTheDocument();
		expect(movie3Title).toBeInTheDocument();
	});

	it('render Banner unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
