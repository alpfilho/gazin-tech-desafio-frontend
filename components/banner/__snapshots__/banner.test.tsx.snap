// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner render Banner unchanged 1`] = `
<div>
  <section
    class="BannerContainer-sc-gyxm3s-0 hlxTxd"
  >
    <button
      class="ControlContainer-sc-ms3p6h-0 LeftControlButton-sc-ms3p6h-1 CYGWr hKylEu"
      data-testid="slider-control"
    >
      <div
        class="ControlIconContainer-sc-ms3p6h-3 eOAwHX"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-left "
          color="#f3f1f1"
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 480c-8.188 0-16.38-3.125-22.62-9.375l-192-192c-12.5-12.5-12.5-32.75 0-45.25l192-192c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L77.25 256l169.4 169.4c12.5 12.5 12.5 32.75 0 45.25C240.4 476.9 232.2 480 224 480z"
            fill="currentColor"
          />
        </svg>
      </div>
    </button>
    <div
      class="BannerContent-sc-gyxm3s-1 gagNOA"
    >
      <div
        class="Container-sc-yq2cgn-0 leaVgY"
      >
        <div
          class="Content-sc-yq2cgn-1 etbrxE"
        >
          <div
            class="RatingContainer-sc-gyxm3s-2 hCuraR"
          >
            <div
              class="RatingContainer-sc-1g1w7oy-0 eRwBjt"
            >
              <div
                class="SvgContainer-sc-1g1w7oy-1 StarContainer-sc-1g1w7oy-2 evdtLQ gZuYIb"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-star "
                  color="#FFB800"
                  data-icon="star"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="RatingText-sc-1g1w7oy-3 dKfZRU"
              >
                <div
                  class="Rating-sc-1g1w7oy-4 iJEgzH"
                >
                  0
                </div>
                <div
                  class="RatingLimit-sc-1g1w7oy-5 kJwsdo"
                >
                  / 10
                </div>
              </div>
              <div
                class="TmbdLogoContainer-sc-1g1w7oy-6 cczYB"
              >
                <svg
                  data-testid="tmdb-logo-primary-short"
                  viewBox="0 0 190.24 81.52"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="imdb-logo-gradient-1"
                      x2="190.24"
                      y1="40.76"
                      y2="40.76"
                    >
                      <stop
                        offset="0"
                        stop-color="#90cea1"
                      />
                      <stop
                        offset=".56"
                        stop-color="#3cbec9"
                      />
                      <stop
                        offset="1"
                        stop-color="#00b3e5"
                      />
                    </lineargradient>
                  </defs>
                  <g
                    data-name="Layer 2"
                  >
                    <path
                      d="M105.67 36.06h66.9a17.67 17.67 0 0 0 17.67-17.66A17.67 17.67 0 0 0 172.57.73h-66.9A17.67 17.67 0 0 0 88 18.4a17.67 17.67 0 0 0 17.67 17.66Zm-88 45h76.9a17.67 17.67 0 0 0 17.67-17.66 17.67 17.67 0 0 0-17.67-17.67h-76.9A17.67 17.67 0 0 0 0 63.4a17.67 17.67 0 0 0 17.67 17.66Zm-7.26-45.64h7.8V6.92h10.1V0h-28v6.9h10.1Zm28.1 0h7.8V8.25h.1l9 27.15h6l9.3-27.15h.1V35.4h7.8V0H66.76l-8.2 23.1h-.1L50.31 0h-11.8Zm113.92 20.25a15.07 15.07 0 0 0-4.52-5.52 18.57 18.57 0 0 0-6.68-3.08 33.54 33.54 0 0 0-8.07-1h-11.7v35.4h12.75a24.58 24.58 0 0 0 7.55-1.15 19.34 19.34 0 0 0 6.35-3.32 16.27 16.27 0 0 0 4.37-5.5 16.91 16.91 0 0 0 1.63-7.58 18.5 18.5 0 0 0-1.68-8.25ZM145 68.6a8.8 8.8 0 0 1-2.64 3.4 10.7 10.7 0 0 1-4 1.82 21.57 21.57 0 0 1-5 .55h-4.05v-21h4.6a17 17 0 0 1 4.67.63 11.66 11.66 0 0 1 3.88 1.87A9.14 9.14 0 0 1 145 59a9.87 9.87 0 0 1 1 4.52 11.89 11.89 0 0 1-1 5.08Zm44.63-.13a8 8 0 0 0-1.58-2.62 8.38 8.38 0 0 0-2.42-1.85 10.31 10.31 0 0 0-3.17-1v-.1a9.22 9.22 0 0 0 4.42-2.82 7.43 7.43 0 0 0 1.68-5 8.42 8.42 0 0 0-1.15-4.65 8.09 8.09 0 0 0-3-2.72 12.56 12.56 0 0 0-4.18-1.3 32.84 32.84 0 0 0-4.62-.33h-13.2v35.4h14.5a22.41 22.41 0 0 0 4.72-.5 13.53 13.53 0 0 0 4.28-1.65 9.42 9.42 0 0 0 3.1-3 8.52 8.52 0 0 0 1.2-4.68 9.39 9.39 0 0 0-.55-3.18Zm-19.42-15.75h5.3a10 10 0 0 1 1.85.18 6.18 6.18 0 0 1 1.7.57 3.39 3.39 0 0 1 1.22 1.13 3.22 3.22 0 0 1 .48 1.82 3.63 3.63 0 0 1-.43 1.8 3.4 3.4 0 0 1-1.12 1.2 4.92 4.92 0 0 1-1.58.65 7.51 7.51 0 0 1-1.77.2h-5.65Zm11.72 20a3.9 3.9 0 0 1-1.22 1.3 4.64 4.64 0 0 1-1.68.7 8.18 8.18 0 0 1-1.82.2h-7v-8h5.9a15.35 15.35 0 0 1 2 .15 8.47 8.47 0 0 1 2.05.55 4 4 0 0 1 1.57 1.18 3.11 3.11 0 0 1 .63 2 3.71 3.71 0 0 1-.43 1.92Z"
                      data-name="Layer 1"
                      fill="url(#imdb-logo-gradient-1)"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </div>
          <a
            class="TextContainer-sc-gyxm3s-3 cedUxo"
            href="/movie-detail/414906"
          >
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 1; transform: translateX(0%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Batman
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Bruce Wayne é um jovem bilionário da cidade de Gotham City, Nova Jersey, que também age secretamente como o vigilante noturno Batman após o assassinato dos seus pais.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Homem-Aranha: Sem Volta Para Casa
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Peter Parker é desmascarado e não consegue mais separar sua vida normal dos grandes riscos de ser um super-herói. Quando ele pede ajuda ao Doutor Estranho, os riscos se tornam ainda mais perigosos, e o forçam a descobrir o que realmente significa ser o Homem-Aranha.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                King's Man: A Origem
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Quando uma série dos piores tiranos e gênios do crime da história se juntam para criar uma guerra que aniquilará milhões, um homem e seu pupilo precisam correr contra o tempo para pará-los.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Agente das Sombras
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Confiança, identidade e o perigo do poder sem controle levam um agente secreto ao limite. Travis Block (Liam Neeson) vive e luta nas sombras. Um “consertador” autônomo do governo, Block é um homem perigoso cujas atribuições incluíam extrair agentes de situações ocultas. Quando Block descobre que um programa sombrio chamado Operação Unidade está abatendo cidadãos comuns por razões conhecidas apenas pelo seu supervisor, o chefe do FBI, Robinson (Aidan Quinn), ele pede a ajuda de uma jornalista (Raver-Lampman), mas seu passado e presente colidem quando seu filha e neta são ameaçadas. Agora Block precisa resgatar as pessoas que ama e expor a verdade para uma chance de redenção. Nada nem ninguém está seguro quando os segredos estão escondidos nas sombras.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Amor, Sublime Amor
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Dois jovens de gangues rivais de Nova York se apaixonam, mas as tensões entre seus respectivos amigos se transformam em tragédia. Remake do filme de 1961.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Pânico
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Vinte e cinco anos após uma série de crimes brutais chocar a tranquila Woodsboro, um novo assassino se apropria da máscara de Ghostface e começa a perseguir um grupo de adolescentes para trazer à tona segredos do passado mortal da cidade.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Contra o Gelo
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Dois homens lutam pela sobrevivência durante uma expedição à Groenlândia em busca de um mapa perdido. Baseado em uma história real ocorrida na Dinamarca em 1909.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Sem Saída
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Darby Thorne é uma mulher que viaja devido a uma emergência familiar, fica presa em nevasca e é forçada a se abrigar numa área de descanso na estrada com um grupo de desconhecidos. Ao topar com uma garota sequestrada numa van, ela encara uma terrível luta de vida ou morte para descobrir quem é o sequestrador.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                Os Piratas: Em Busca do Tesouro Perdido
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Durante a Dinastia Joseon, uma corajosa tripulação de piratas e bandidos luta contra adversidades, pistas intrigantes e rivais em busca do ouro real perdido no mar.
              </p>
            </div>
            <div
              class="TextContentContainer-sc-421v6c-0 gQLyUw"
              style="opacity: 0; transform: translateX(25%) translateZ(0);"
            >
              <h1
                class="Title-sc-12jcw8v-0 Title-sc-421v6c-1 gtTvjR gctaaJ"
              >
                O Massacre da Serra Elétrica: O Retorno de Leatherface
              </h1>
              <p
                class="Synopsis-sc-421v6c-2 jooTuS"
              >
                Nesta sequência, um grupo de jovens decide dar vida nova a uma cidade fantasma no Texas, mas acaba se deparando com Leatherface, o famoso maníaco da serra elétrica.
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="BackgroundContainer-sc-gyxm3s-4 fBiith"
    >
      <div
        class="BackgroundContent-sc-gyxm3s-5 VBrVU"
      >
        <img
          class="BackgroundImage-sc-gyxm3s-6 jHCUBt"
          style="opacity: 1; z-index: 1;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 eYRbTY"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 OeQqR"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 iMlHZn"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 kakdtI"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 gmGwhK"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 jlZyVs"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 khHdVz"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 dXQzZ"
          style="opacity: 0; z-index: 0;"
        />
        <img
          class="BackgroundImage-sc-gyxm3s-6 dimddI"
          style="opacity: 0; z-index: 0;"
        />
        <div
          class="BackgroundOverlay-sc-gyxm3s-7 eMXPem"
        />
        <div
          class="TextBackground-sc-gyxm3s-8 dAeepP"
        />
      </div>
    </div>
    <button
      class="ControlContainer-sc-ms3p6h-0 RightControlButton-sc-ms3p6h-2 CYGWr bQGCVZ"
      data-testid="slider-control"
    >
      <div
        class="ControlIconContainer-sc-ms3p6h-3 eOAwHX"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-right "
          color="#f3f1f1"
          data-icon="chevron-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 480c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L242.8 256L73.38 86.63c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25l-192 192C112.4 476.9 104.2 480 96 480z"
            fill="currentColor"
          />
        </svg>
      </div>
    </button>
  </section>
</div>
`;
